internal/dag/cache.go

```diff
@@ -511,38 +513,15 @@ func (kc *KubernetesCache) secretTriggersRebuild(secret *v1.Secret) bool {
 		return true
 	}
 
-	delegations := make(map[string]bool) // targetnamespace/secretname to bool
-
-	// TODO(youngnick): Check if this is required.
-	for _, d := range kc.tlscertificatedelegations {
-		for _, cd := range d.Spec.Delegations {
-			for _, n := range cd.TargetNamespaces {
-				delegations[n+"/"+cd.SecretName] = true
-			}
-		}
+	secret := types.NamespacedName{
+		Namespace: secretObj.Namespace,
+		Name:      secretObj.Name,
 	}
 
 	for _, ingress := range kc.ingresses {
-		if ingress.Namespace == secret.Namespace {
-			for _, tls := range ingress.Spec.TLS {
-				if tls.SecretName == secret.Name {
-					return true
-				}
-			}
-		}
-		if delegations[ingress.Namespace+"/"+secret.Name] {
-			for _, tls := range ingress.Spec.TLS {
-				if tls.SecretName == secret.Namespace+"/"+secret.Name {
-					return true
-				}
-			}
-		}
-
-		if delegations["*/"+secret.Name] {
-			for _, tls := range ingress.Spec.TLS {
-				if tls.SecretName == secret.Namespace+"/"+secret.Name {
-					return true
-				}
+		for _, tls := range ingress.Spec.TLS {
+			if secret == k8s.NamespacedNameFrom(tls.SecretName, k8s.TLSCertAnnotationNamespace(ingress)) {
```

Comment:
Still need to fall back to the ingress's namespace as a last resort:
```suggestion
			if secret == k8s.NamespacedNameFrom(tls.SecretName, k8s.TLSCertAnnotationNamespace(ingress),  k8s.DefaultNamespace(ingress.Namespace)) {
```

Reply:
This was first time I dared to try click "commit suggestion" but I failed :-D

Reply:
Sorry, my suggestion was invalid I guess! Looks like extra space